sdk/core/System.ClientModel/src/Message/BinaryContent.cs

```diff
@@ -65,6 +72,61 @@ public static BinaryContent Create(Stream stream)
         return new StreamBinaryContent(stream);
     }
 
+    /// <summary>
+    /// Creates an instance of <see cref="BinaryContent"/> that contains the
+    /// JSON representation of the provided object.
+    /// </summary>
+    /// <typeparam name="T">The type of the object to serialize.</typeparam>
+    /// <param name="jsonSerializable">The object to serialize to JSON.</param>
+    /// <param name="options">The <see cref="JsonSerializerOptions"/> to use for serialization.
+    /// If not provided, the default options will be used.</param>
+    /// <returns>An instance of <see cref="BinaryContent"/> that contains the
+    /// JSON representation of the provided object.</returns>
+#pragma warning disable AZC0014 // Avoid using banned types in public API
+    public static BinaryContent CreateJson<T>(T jsonSerializable, JsonSerializerOptions? options = default)
+#pragma warning restore AZC0014 // Avoid using banned types in public API
+    {
+        Argument.AssertNotNull(jsonSerializable, nameof(jsonSerializable));
+
+        BinaryData data = BinaryData.FromObjectAsJson(jsonSerializable, options);
+        return new BinaryDataBinaryContent(data.ToMemory(), "application/json");
+    }
+
+    /// <summary>
+    /// Creates an instance of <see cref="BinaryContent"/> that contains the
+    /// JSON representation of the provided object using the specified JSON type information.
+    /// </summary>
+    /// <typeparam name="T">The type of the object to serialize.</typeparam>
+    /// <param name="jsonSerializable">The object to serialize to JSON.</param>
+    /// <param name="jsonTypeInfo">The <see cref="JsonTypeInfo{T}"/> to use for serialization.</param>
+    /// <returns>An instance of <see cref="BinaryContent"/> that contains the
+    /// JSON representation of the provided object.</returns>
+#pragma warning disable AZC0014 // Avoid using banned types in public API
+    public static BinaryContent CreateJson<T>(T jsonSerializable, JsonTypeInfo<T> jsonTypeInfo)
+#pragma warning restore AZC0014 // Avoid using banned types in public API
+    {
+        Argument.AssertNotNull(jsonSerializable, nameof(jsonSerializable));
+        Argument.AssertNotNull(jsonTypeInfo, nameof(jsonTypeInfo));
+
+        BinaryData data = BinaryData.FromObjectAsJson(jsonSerializable, jsonTypeInfo);
+        return new BinaryDataBinaryContent(data.ToMemory(), "application/json");
+    }
+
+    /// <summary>
+    /// Creates an instance of <see cref="BinaryContent"/> that contains the
+    /// provided JSON string.
+    /// </summary>
+    /// <param name="jsonString">The JSON string to be used as the content.</param>
+    /// <returns>An instance of <see cref="BinaryContent"/> that contains the
+    /// provided JSON string.</returns>
+    public static BinaryContent CreateJson(string jsonString)
```

Comment:
let's add an optional boolean to validate that the string is actually json. Let's use Utf8JsonReader to do the validation